contract/Portkey.Contracts.CA/CAContract_Actions.cs

```diff
@@ -321,4 +414,109 @@ public override GetCheckOperationDetailsInSignatureEnabledOutput
+        return State.CircuitIdToVerifyingKey[input.Value];
+    }
+
+    public override Empty StartOracleRequest(StartOracleRequestInput input)
+    {
+        Assert(input != null, "Invalid input.");
+        Assert(input.SubscriptionId > 0, "Invalid input subscription id.");
+        Assert(input.RequestTypeIndex > 0, "Invalid request type index.");
+
+        State.OracleContract.SendRequest.Send(new SendRequestInput
+        {
+            SubscriptionId = input.SubscriptionId,
+            RequestTypeIndex = input.RequestTypeIndex,
+            SpecificData = input.SpecificData
+        });
+        
+        return new Empty();
+    }
+
+    public override Empty HandleOracleFulfillment(HandleOracleFulfillmentInput input)
+    {
+        Assert(input != null, "Invalid input.");
+        // Assert(IsHashValid(input.RequestId), "Invalid input request id.");
+        Assert(input.RequestTypeIndex > 0, "Invalid request type index.");
+        Assert(!input.Response.IsNullOrEmpty() || !input.Err.IsNullOrEmpty(), "Invalid input response or err.");
+        GoogleResponse reponse = JsonConvert.DeserializeObject<GoogleResponse>(System.Text.Encoding.UTF8.GetString(input.Response.ToByteArray()));
+        Assert(reponse != null, "Invalid input.");
+        //todo aetherlink can't differentiate apple/google/facebook
+        foreach (var googleKeyDto in reponse.Keys)
+        {
+            State.KidToPublicKey[GuardianType.OfGoogle][googleKeyDto.Kid] = googleKeyDto.N;
+        }
+
+        return new Empty();
+    }
+    
+    // {
+    //     "keys": [
+    //     {
+    //         "n": "w-l_VE4KNa22n4nsMwcabujowm924YoQQnwOz_dPYHmDI1O-r2bqw6mHmByXwii7aaeIMHJZWpmT5SkR3OYIu5RbSgiU-8JrQoplW_vZY2IqG1y5-frPC_9gnz_0qKKjtjqglCP-1AlfOdu7r5kOpkOACs5mWn4tm1K9R1EPjk2T_MMO7FkteZd8woh1fwUUuvbhPyDxBzx9EUsnGWbpTndOYc7W-EUk1jMtWBk3buLeaypVaOLWranK_XFrX-xx03BohrfinOqmftYgc0z94sxix7X1G36JZeh8-jpUhwyBPinBxOZOE_5kQn4CYM66Ygxwiws0ZJ-klG2qTi239w",
+    //         "kid": "674dbba8faee69acae1bc1be190453678f472803",
+    //         "e": "AQAB",
+    //         "alg": "RS256",
+    //         "kty": "RSA",
+    //         "use": "sig"
+    //     },
+    //     {
+    //         "e": "AQAB",
+    //         "alg": "RS256",
+    //         "kid": "c3abe413b2268ae976458c82c15179547e97527e",
+    //         "kty": "RSA",
+    //         "use": "sig",
+    //         "n": "rr9RI_trqotRBIi9qTgoYPE4FnjvhIOoolHhi3Lxw3FqK-8vYwczsVEsTzVek3MAUdNNILhJozKN1Snirlrr_albKitJFJ1SbjJe2HsNeZu96TrWMNmWCCUOXK0ecEVdw8gqfGOE-_VvKaIVxYGaJpSystRjRXfrT8QHXNYMSI7xqJI9Obw4IUpwYxky6eVtB3zLW4Qz7cH2jFMsSB3uufdvOA-odJXrwZkC2FX0krVtSyCWzazjHMX0zOCckZVZK7xBNjuuElcnZ4IpGHcHDYtyI004WY-ez7_yxyOHGoJd31Omg39DkunpSQxfA87LLGumtd1OROvQzNFZETDqfw"
+    //     }
+    //     ]
+    // }
+    
+    internal class GoogleResponse
+    {
+        public List<GoogleKeyDto> Keys { get; set; }
+    }
+    
+    internal class GoogleKeyDto
```

Comment:
This is not specific to Google. Other providers follow the same.

https://www.facebook.com/.well-known/oauth/openid/jwks/?paipv=0&eav=AfZoWUgLyOKRPWYYbs05wEzwNngEk-s1_3Kx5sO3REdP_fCScnpPbBveGmTOF5FdgYg&_rdr

https://appleid.apple.com/auth/keys